modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -2597,8 +2615,17 @@ private GridTuple3<Long, Long, Boolean> ttlAndExpireTime(IgniteCacheExpiryPolicy
      */
     private void drReplicate(GridDrType drType, @Nullable CacheObject val, GridCacheVersion ver, AffinityTopologyVersion topVer)
         throws IgniteCheckedException {
-        if (cctx.isDrEnabled() && drType != DR_NONE && !isInternal())
+        if (cctx.isDrEnabled() && drType != DR_NONE && !isInternal()) {
+            if (log.isDebugEnabled())
+                log.debug("Key replication, key=" + keyValue(false) + ", dr=" + cctx.isDrEnabled()
+                        + ", drType=" + drType);
+
             cctx.dr().replicate(key, val, rawTtl(), rawExpireTime(), ver.conflictVersion(), drType, topVer);
+        } else {
+            if (log.isDebugEnabled())
+                log.debug("Skipping key replication, key=" + keyValue(false) + ", dr=" + cctx.isDrEnabled()
+                        + ", drType=" + drType);
+        }
```

Comment:
It make no sense to log smth if DR is disabled.
Let's log at trace level.
Let's respect data sensitivity.